### ISSUE TITLE
Not all dependencies copied to build server...

### ISSUE THREAD

**Issue by groskopf** [mozilla/sccache#1076]
I have the sccache distibuted build server setup where I am not able to get the toolchain working correctly. I have identified the root cause to be related to the fact that ../lib/libdl.so.2 is not copied to the build server toolchain cache and the compiler cannot start. The toolchain is generated by yocto and install in the docker from which i try to use sccache.

cut out of compilation error message:
```
 sccache /opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/bin/arm-dse13mk2-linux-gnueabi/arm-dse13mk2-linux-gnueabi-g++   -march=armv5te -marm -fstack-protector-strong  -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/opt/dse13mk2-os/0.1/sysroots/armv5e-dse13mk2-linux-gnueabi --sysroot=/opt/dse13mk2-os/0.1/sysroots/armv5e-dse13mk2-linux-gnueabi  -DDSE13MK2_WEBUI_VERSION=\"3.0.496\" -DSHOW_MESSAGE=1 -I_deps/jwt-cpp-build/src -I_deps/jwt-cpp-src/src -I_deps/jwt-cpp-src/src/include -I_deps/jwt-cpp-src/src/include/private -O2 -pipe -g -feliminate-unused-debug-types  -Wall   -std=gnu++11 -MD -MT _deps/jwt-cpp-build/src/CMakeFiles/jwt.dir/validators/hmacvalidator.cpp.o -MF _deps/jwt-cpp-build/src/CMakeFiles/jwt.dir/validators/hmacvalidator.cpp.o.d -o _deps/jwt-cpp-build/src/CMakeFiles/jwt.dir/validators/hmacvalidator.cpp.o -c _deps/jwt-cpp-src/src/validators/hmacvalidator.cpp
/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/cc1plus: error while loading shared libraries: libdl.so.2: cannot open shared object file: No such file or directory
/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/as: error while loading shared libraries: libdl.so.2: cannot open shared object file: No such file or directory
sccache: Compiler killed by signal 1
```

"chrpath" tells me that inside the local docker I have path to the libdl.so.2:

```
mydocker$chrpath -l /opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/cc1plus
/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/cc1plus: RPATH=$ORIGIN/../../../../../lib:$ORIGIN/../../../../../../lib
 ```
```
mydocker$ ll /opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/../../../../../../lib | grep libdl.so.2
lrwxrwxrwx 1 root root      13 Nov 29 08:52 libdl.so.2 -> libdl-2.31.so
```
And the the toolchain is also working fine here. 

On the build server the library is then missing:

```
buildserver$ ll /tmp/build/toolchains/8a7676534538ac1dc7549e4034a0b16a3d901db3b837a4f881a9032294d0dd5b/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/lib
total 3264
drwxr-xr-x 2 root root    4096 Dec  1 08:38 ./
drwxr-xr-x 4 root root    4096 Dec  1 08:38 ../
-rwxr-xr-x 1 root root  186152 Nov 29 08:52 ld-linux-x86-64.so.2*
-rwxr-xr-x 1 root root 1822968 Nov 29 08:52 libc.so.6*
-rwxr-xr-x 1 root root 1316896 Nov 29 08:52 libm.so.6*
```
And can see the rpath is not matching what I have in the client docker:
```
buildserver$ chrpath -l /tmp/build/toolchains/8a7676534538ac1dc7549e4034a0b16a3d901db3b837a4f881a9032294d0dd5b/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/cc1plus
/tmp/build/toolchains/8a7676534538ac1dc7549e4034a0b16a3d901db3b837a4f881a9032294d0dd5b/opt/dse13mk2-os/0.1/sysroots/x86_64-dse13sdk-linux/usr/libexec/arm-dse13mk2-linux-gnueabi/gcc/arm-dse13mk2-linux-gnueabi/9.3.0/cc1plus: RPATH=$ORIGIN/../../../../../lib
```
Here there is only one path of the paths. This actually match how the yocto setup the toolchain before I patch it with the extra rpath to the /lib/libdl.so.2  

To sum up: I can see that libdl.so.2 is not copied to the toolchaing cache. Properly related to the rpath some how is altered/not present in the compiler in the build server toolchain cache.

**Comment by groskopf** [comment]
This has been solve by not using release version v0.2.15 but building a later version.
